### PR TITLE
Deprecate devbox build and plan commands

### PR DESCRIPTION
**Pull request by mohsenari**
## Summary
- Deprecated `devbox build` and `devbox plan` commands
- Removed Build plans from planners
- Removed build from tests
## How was it tested?
- Compiles
- `devbox build` and `plan` show this output:
```sh
$ ./devbox build
Command "build" is deprecated, Please follow devbox documentation on how to build a container image around your devbox project.
$ ./devbox plan
Command "plan" is deprecated, Please follow devbox documentation on how to build a container image around your devbox project.
```

### CODE REVIEW

File: boxcli/plan.go [hunk 1]
```diff
@@ -20,46 +15,15 @@ func PlanCmd() *cobra.Command {
...
-		Use:   "plan",
-		Short: "Preview the plan used to build your environment",
-		Args:  cobra.MaximumNArgs(1),
+		Use:        "plan",
+		Short:      "Preview the plan used to build your environment",
+		Deprecated: "Please follow devbox documentation on how to build a container image around your devbox project.",
+		Args:       cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runPlanCmd(cmd, args, flags)
+			return nil
```
Comment (LucilleH) [inline boxcli/plan.go:23]: `plan` command is good for testing shell stuff as well. Can we simply hide instead of removing it?

**Review by LucilleH — APPROVED** [review]

**Comment by LucilleH** [comment]
Do we have documentation on how to build a container?

**Comment by mohsenari** [comment]
> Do we have documentation on how to build a container?

No, that was my next task to create documentation or at least some general guidelines on building a container.

File: boxcli/plan.go (hunk 1 above)
Comment (mohsenari) [reply boxcli/plan.go:23]: Fair! I didn't know it's used for testing. I'll update the PR.

**Comment by LucilleH** [comment]
We probably need to do a pass on the docs, since it still has quite a bit of `devbox build` related stuff there.

File: boxcli/plan.go (hunk 1 above)
Comment (mohsenari) [reply boxcli/plan.go]: @LucilleH This pr is too convoluted to keep `plan` and remove the build. I'll close it and open a new PR in a different branch

**Comment by mohsenari** [comment]
closing this in favor of #336

**Pull request closed without merging**